### ISSUE TITLE
[Linux] Migrate to the new update semantics embedder API

### ISSUE THREAD

**Issue by loic-sharma** [flutter/flutter#132517]
The original update semantics embedder API was deprecated (see https://github.com/flutter/flutter/issues/114201) and replaced by a [new update semantics embedder API](https://github.com/flutter/engine/blob/785b24900a360ed872d1f779a65d6c062ce69448/shell/platform/embedder/embedder.h#L2110-L2120). The Linux embedder should be migrated to the new embedder API.

The [Windows](https://github.com/flutter/engine/blob/785b24900a360ed872d1f779a65d6c062ce69448/shell/platform/windows/flutter_windows_engine.cc#L345-L369) and [macOS](https://github.com/flutter/engine/blob/785b24900a360ed872d1f779a65d6c062ce69448/shell/platform/darwin/macos/framework/Source/FlutterEngine.mm#L544-L551) embedders have already been migrated.

/cc @robert-ancell @jpnurmi

**Comment by robert-ancell** [comment]
Was fixed in https://github.com/flutter/engine/pull/51030
